pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildpacks
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/mount"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+func (b *BuildpackBuilder) build(ctx context.Context, out io.Writer, workspace string, artifact *latest.BuildpackArtifact, latest string) error {
+	builderImage := artifact.Builder
+	logrus.Debugln("Builder image", builderImage)
+	if err := b.pull(ctx, out, builderImage, artifact.ForcePull); err != nil {
+		return err
+	}
+
+	runImage, err := b.runImage(ctx, artifact, builderImage)
+	if err != nil {
+		return err
+	}
+	logrus.Debugln("Run image", runImage)
+	if err := b.pull(ctx, out, runImage, artifact.ForcePull); err != nil {
+		return err
+	}
+
+	cacheID := fmt.Sprintf("%.6x", sha256.Sum256([]byte(latest)))
+	workspaceVol := fmt.Sprintf("pack-%s.workspace", cacheID)
+	layersVol := fmt.Sprintf("pack-%s.layers", cacheID)
+	launchCacheVol := fmt.Sprintf("pack-cache-%s.launch", cacheID)
+	buildCacheVol := "pack-cache-skaffold.build"
+
+	logrus.Debugln("Get dependencies")
+	deps, err := GetDependencies(ctx, workspace, artifact)
+	if err != nil {
+		return err
+	}
+
+	var paths []string
+	for _, dep := range deps {
+		paths = append(paths, filepath.Join(workspace, dep))
+	}
+
+	copyWorkspace := func(ctx context.Context, container string) error {
+		return b.localDocker.CopyToContainer(ctx, container, "/workspace", workspace, paths)
+	}
+
+	detector := docker.ContainerRun{
+		Image:       builderImage,
+		Command:     []string{"/lifecycle/detector"},
+		BeforeStart: copyWorkspace,
+		Mounts: []mount.Mount{
+			{Type: mount.TypeVolume, Source: workspaceVol, Target: "/workspace"},
+			{Type: mount.TypeVolume, Source: layersVol, Target: "/layers"},
+		},
+	}
+	restorerAnalyzer := docker.ContainerRun{
+		Image:   builderImage,
+		Command: []string{"sh", "-c", "/lifecycle/restorer -path /cache && /lifecycle/analyzer -daemon " + latest},
```

Comment:
will these commands ever be something that users will want to configure? i'm imagining extra args that we expose through the `skaffold.yaml` that get passed in here.

Reply:
Yes, at some point, users will at least be able to pass env variables.